### PR TITLE
add pixel events

### PR DESCRIPTION
**Pull request by hiagolcm**
Send pixel messages on autocomplete interaction

```
"product_click",
"search_suggestion_click",
"top_search_click",
"history_click",
"search",
"empty_search",
"see_all_click",
```

[workspace](https://hiago--eriksbikeshop.myvtex.com/bike?map=ft)

### CODE REVIEW

File: react/utils/withUsePixel.tsx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+import { usePixel } from "vtex.pixel-manager/PixelContext";
+
+// tslint:disable-next-line:variable-name
+export default function withUsePixel(ComponentWrapped: typeof React.Component) {
```
Comment (chrsmutti) [inline react/utils/withUsePixel.tsx:5]: There is a `withPixel` implementation on pixel-manager that we can use:
https://github.com/vtex/pixel-manager/blob/master/react/PixelContext.tsx#L67-L79

File: react/components/Autocomplete/index.tsx [hunk 2]
```diff
@@ -265,6 +327,23 @@ class AutoComplete extends React.Component<
...
+        event: EVENT_NAME,
+        eventType: count > 0 ? EventType.Search : EventType.EmptySearch,
+        search: {
+          operator,
+          misspelled,
+          text: decodeURI(term),
+          match: count,
+        },
+      });
+    }
```
Comment (chrsmutti) [inline react/components/Autocomplete/index.tsx:345]: What if we move this, and the `handle*Click` functions to a separate file? Will it be weird?
As this one file has 500+ lines, maybe dividing it into more specific files would help a bit with readability.

If you opt to go this way the functions would probably look like:

```typescript
export function handleProductClick(push: (data: PixelData) => void, page: string, productId: string, position: number): void
// or maybe
export function buildProductClickHandler(push: (data: PixelData) => void, page: string): (string, number) => void
```

It's really subjective, as we would be removing it from the component. A different approach would be a Provider with this functions:

```jsx
<AutocompleteEventProvider handleProductClick={handleProductClick} handleItemClick={handleItemClick} /* ... */ />
```

**Review by chrsmutti — CHANGES_REQUESTED** [review]

File: react/utils/withUsePixel.tsx (hunk 1 above)
Comment (hiagolcm) [reply react/utils/withUsePixel.tsx]: Done!

File: react/components/Autocomplete/index.tsx (hunk 2 above)
Comment (hiagolcm) [reply react/components/Autocomplete/index.tsx]: You're right. This file is huge.

I'm keeping the handle functions on the `utils/pixel.ts` file now. Is it better?

**Review by chrsmutti — APPROVED** [review]
LGTM! 🎉

**Pull request merged**